modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py

```diff
@@ -122,7 +122,8 @@ def wait_partitions(cls, partitions):
             NumPy array with ``PandasDataframePartition``-s.
         """
         ray.wait(
-            [partition._data for partition in partitions], num_returns=len(partitions)
+            [partition.list_of_blocks[0] for partition in partitions],
```

Comment:
We should wait on all the blocks. Probably defining `blocks = [block for partition in partitions for block in partition.list_of_blocks]` separately so we can get its length for `num_returns`.

Reply:
fixed